engine/gamesys/src/gamesys/components/comp_sprite.cpp

```diff
@@ -565,9 +572,17 @@ namespace dmGameSystem
         SpriteResource* resource = first->m_Resource;
         TextureSetResource* texture_set = GetTextureSet(first, resource);
 
-        // Ninja in-place writing of render object.
-        dmRender::RenderObject& ro = *sprite_world->m_RenderObjects.End();
-        sprite_world->m_RenderObjects.SetSize(sprite_world->m_RenderObjects.Size()+1);
+        // Although we generally like to preallocate it, we cannot since we
+        // 1) don't want to preallocate max_sprite number of render objects and
+        // 2) We cannot keep the render object in a (small) fixed array and then reallocate it, since we pass the pointer to the render engine
+        if (sprite_world->m_RenderObjectsInUse == sprite_world->m_RenderObjects.Capacity())
```

Comment:
We just let it grow dynamically, so that we allocate a minimum amount of objects.